libs/blocks/aside/aside.js

```diff
@@ -105,11 +105,61 @@ function decorateBlockBg(block, node) {
   }
 }
 
+function addPromobar(sourceEl, parent) {
+  const newPromo = sourceEl.cloneNode(true);
+  parent.appendChild(newPromo);
+}
+
+function checkViewportPromobar(foreground) {
+  const childCount = foreground.childElementCount;
+  const { children } = foreground;
+  if (childCount < 2) addPromobar(children[childCount - 1], foreground);
+  if (childCount < 3) addPromobar(children[childCount - 1], foreground);
+}
+
+function combineTextBocks(textBlocks, iconArea, viewPort) {
+  const textStyle = viewPort === 'desktop-up' ? ['m', 'l'] : ['s', 's'];
+  const contentArea = createTag('p', { class: 'content-area' });
+  const textArea = createTag('p', { class: 'text-area' });
+  textBlocks[0].parentElement.prepend(contentArea);
+  textBlocks.forEach((textBlock) => {
+    textArea.appendChild(textBlock);
+    if (textBlock.nodeName === 'P') {
+      textBlock.classList.add(`body-${textStyle[1]}`);
+    } else {
+      textBlock.classList.add(`heading-${textStyle[0]}`);
+    }
+  });
+  if (iconArea) {
+    iconArea.classList.add('icon-area');
+    contentArea.appendChild(iconArea);
+  }
+  contentArea.appendChild(textArea);
+}
+
+function decoratePromobar(el) {
+  const viewports = ['mobile-up', 'tablet-up', 'desktop-up'];
+  const foreground = el.querySelector('.foreground');
+  if (foreground.childElementCount !== 3) checkViewportPromobar(foreground);
```

Comment:
Instead of duplicating elements can you instead add multiple `-up` classes when the content is the same for all widths?

E.g: `class="mobile-up tablet-up desktop-up"`

It does look like a few css tweaks will be required for this

Reply:
@chrischrischris We are duplicating the content as the styling as per the XD in promo aside is different for the content area across viewports. 
For instance the Heading is heading-m in desktop but heading-s in tablet. Hence we are creating viewport specific text areas and adding specifying styling options to it. To control the changing styling with media screen change would require additional JS to monitor the viewport change.